app.py

```diff
@@ -1,1268 +1,174 @@
-        authors_full_name = []
-        for i in authors:
-            authors_full_name.append(people_data[i]["name"])
-        record["authors"] = authors_full_name
-        outr_d = {}
-        pkey = ""
-        title_upper = record["title"].upper()
-        for i in title_upper.split(" "):
-            pkey += i[0]
-        outr_d[pkey] = record
-        
-        with open("solarillion.github.io/_data/publications.yml", "a") as file:
-                dump(outr_d, file, width=1000)
-        
-        branch_name = 'add-publication-' + pkey
-        commit_message = "publications.yml: added publication"
-        pr_title = "Added new publication"
-        pr_body = "Added a new publication: \"" + record["title"] + "\"."
 
-        git_push(full_local_path, branch_name, commit_message, pr_title, pr_body)
-        
-        data1 = {}
-        data1["status"] = "The publication will be added to the website."
-        return render_template("login.html",data=data1) 
-    except Exception as e:
-        message = "An error occurred while adding publication: " + str(request.form['pname'])
-        message += "\nException:\n" + str(e)
-        tars.chat_postMessage(channel=tars_admin, text=message)
+@app.message("ping")
+def message_hello(message, say):
+    say("pong")
 
 
-@app.route("/add-person", methods=['GET', 'POST'])
-@flask_login.login_required
-def addperson():
-    if request.method == 'GET':
-        return render_template('add-person.html')
- 
-    try:
-        data = request.form
-        img_file = request.files['file']
-        full_local_path = "solarillion.github.io"
-        git_clone(full_local_path)
+@app.message("request office hours")
+def request_office_hours(message, say):
+    admin = list(db.child(key_fb_tars).child("admin").get().val())
+    if message["user"] not in admin:
+        say("You're not allowed to do this!")
+        return
 
-        person = data['name'].replace(" ","") + ":\n"
-        person += '  group: ra\n'
-        person += '  avatar: /assets/images/headshots/' + data['name'].replace(" ","") +\
-            '.' + img_file.filename.split('.')[-1].lower() + "\n"
-        person += '  page: /people/' + data['name'].replace(" ","") + "\n"
-        for key in data.keys():
-            if key == 'bio':
-                roles = data.getlist(key)
-                temp_str = ', '.join(roles)
-                person += '  bio: ' + temp_str + '\n'
-            elif key == 'about':
-                person += '  about:\n    \"' + data.get(key) + '\"\n\n'
-            else:
-                value = data.get(key)
-                if len(value) > 0:
-                    person += '  ' + key + ': ' + value + '\n'
-        
-        with open("solarillion.github.io/_data/people.yml", 'at') as file:
-            file.write(person)
+    app.client.chat_postMessage(
+        channel=vineethv_id,
+        text="Sir, please fill your office hours in this form: https://forms.gle/eMoayTXg5KJCata68",
```

Comment:
These URLs shouldn't be public and should be stored in an environment variable instead. (Note: We should change this URL.)